terraform/modules/sharded_module/simple_deployment.tf

```diff
@@ -0,0 +1,165 @@
+locals {
+  mongodb_apps = {
+    "config-server" = {
+      app_name = var.config_server_app_name
+      units    = var.config_server_units
+      role     = "config-server"
+    }
+    "shard-one" = {
+      app_name = var.shard_one_app_name
+      units    = var.shard_one_units
+      role     = "shard"
+    }
+    "shard-two" = {
+      app_name = var.shard_two_app_name
+      units    = var.shard_two_units
+      role     = "shard"
+    }
+  }
```

Comment:
Noob question: If someone wants more shards, they have to edit this file?
There is no way to provide an array of shards instead?

Reply:
Correct, they would have to edit this file. I am not sure if it is possible to provide an array in this case or not.

Reply:
Neha's right, I think here we want to add a `var` in `variables.tf`, let's call it 'topology' or something:

```tf
variable "topology" {
  description = "mongo large deployment topology"
  type        = optional(list(map(string)), {})
  default     = [
    	"config-server" = {
      		app_name = var.config_server_app_name
      		units    = var.config_server_units
      		role     = "config-server"
    	}
}
```

This is really messy, but hopefully you get the gist.
Useful examples here - https://discuss.hashicorp.com/t/why-not-arrays-of-objects/54481/4

Reply:
TIL! Thank you both!